addons/binding/org.openhab.binding.smappee/src/main/java/org/openhab/binding/smappee/internal/discovery/SmappeeDiscoveryService.java

```diff
@@ -0,0 +1,201 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.smappee.internal.discovery;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.smappee.SmappeeBindingConstants;
+import org.openhab.binding.smappee.handler.SmappeeHandler;
+import org.openhab.binding.smappee.internal.SmappeeServiceLocationInfo;
+import org.openhab.binding.smappee.internal.SmappeeServiceLocationInfoActuator;
+import org.openhab.binding.smappee.internal.SmappeeServiceLocationInfoAppliance;
+import org.openhab.binding.smappee.internal.SmappeeServiceLocationInfoSensor;
+import org.openhab.binding.smappee.internal.SmappeeServiceLocationInfoSensorChannel;
+import org.openhab.binding.smappee.internal.ThingProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableSet;
+
+/**
+ * Discovery class for the Smappee.
+ * This will detect the switches in your home.
+ *
+ * @author Niko Tanghe - Initial contribution
+ */
+public class SmappeeDiscoveryService extends AbstractDiscoveryService {
```

Comment:
ok

Reply:
I think my comment was not correct, since you manually register the discovery service in the bridge and thus this annotation is not needed. The use of the annotation is when it runs independent of the bridge(handler), but that might require a different implementation of the discovery service. I don't know if that is possible.